euphonic/readers/castep.py

```diff
@@ -458,19 +442,17 @@ def read_interpolation_data(
     with open(filename, 'rb') as f:
         int_type = '>i4'
         float_type = '>f8'
-        header = ''
         first_cell_read = True
-        while header.strip() != b'END':
-            header = _read_entry(f)
-            if header.strip() == b'BEGIN_UNIT_CELL':
+        while (header := _read_entry(f).strip()) != b'END':
+            if header == b'BEGIN_UNIT_CELL':
```

Comment:
Although this might be something you want to split into functions and have a `dict[bytes, Callable]` to select? Possibly overcomplicating things though

Reply:
The reason I prefer match is it explicitly says what variable is being checked. Same reason why I have always wanted one rather than `elif` chains.

```
if i == a:
elif i == b:
elif l == c:
elif i == d:
else:
```
Stupidly contrived example, but in the middle is an `l` not an `i` which in a huge chain can be missed. Of course this is bad practice, but I've seen that kind of thing before in huge lists.

Reply:
That makes sense, one less thing to look for in each block.

Are we agreed that for now breaking this up would make it worse?

Reply:
Yeah, it was just something to consider.